rocAL/source/readers/image/cifar10_data_reader.cpp

```diff
@@ -53,11 +56,18 @@ Reader::Status CIFAR10DataReader::initialize(ReaderConfig desc) {
     _shard_size = _sharding_info.shard_size;
     _shuffle = desc.shuffle();
     ret = subfolder_reading();
-    _curr_file_idx = _shard_start_idx_vector[_shard_id]; // shard's start_idx would vary for every shard in the vector
     // shuffle dataset if set
-    if (ret == Reader::Status::OK && _shuffle)
-        std::random_shuffle(_file_names.begin() + _shard_start_idx_vector[_shard_id],
-                            _file_names.begin() + _shard_end_idx_vector[_shard_id]);
+    if (ret == Reader::Status::OK && _shuffle) {
+        std::mt19937 rng1(_shard_id);
+        auto rng2 = rng1;
```

Comment:
what is the differene between new rng2 and rng3 Vs using rng1 for all 3?

Reply:
Each time the rng1 is called internally inside std::shuffle, its internal state is changed hence it won't give the same random value for the second std::shuffle. Hence I created two more rng variables with the same initial state as rng1 and uses them in std::shuffle to get the same shuffling order

Reply:
In that case, you want to give a random seed to the generator like std::random_device  and use the same random number generator. Assigning one to another one is not recommended

Reply:
Seeding the std::mt19937 instance with the same random seed still gives different values for each vector. Shuffling the indices requires a temporary vector to store shuffled elements and an indices vector increasing the memory usage